src/main/kotlin/no/nb/mlt/wls/order/service/OrderService.kt

```diff
@@ -65,6 +67,53 @@ class OrderService(val db: OrderRepository, val synqService: SynqOrderService) {
         return ResponseEntity.status(HttpStatus.CREATED).body(order.toApiOrderPayload())
     }
 
+    suspend fun updateOrder(payload: ApiOrderPayload): ResponseEntity<ApiOrderPayload> {
+        throwIfInvalidPayload(payload)
+
+        val existingOrder =
+            db.findByHostNameAndHostOrderId(payload.hostName, payload.orderId)
+                .timeout(Duration.ofSeconds(8))
+                .onErrorMap {
+                    if (it is TimeoutException) {
+                        logger.error(it) {
+                            "Timed out while fetching from WLS database. Relevant payload: $payload"
+                        }
+                    } else {
+                        logger.error(it) { "Unexpected error for $payload" }
+                    }
+                    ServerErrorException("Failed while checking if order already exists in the database", it)
+                }
+                .awaitSingleOrNull()
+
+        if (existingOrder == null) {
+            throw ResponseStatusException(
+                HttpStatus.BAD_REQUEST,
+                "Order with id $payload.hostOrderId from $payload.hostName does not exist in the database"
+            )
+        }
+
+        if (existingOrder.status != OrderStatus.NOT_STARTED) {
+            throw ResponseStatusException(HttpStatus.CONFLICT, "Order is already being processed, and can not be edited")
+        }
+        synqService.updateOrder(payload)
+
+        // Saving here will override the existing order, as the id's match
+        val updatedOrder =
+            db.save(
+                existingOrder.copy(
+                    hostOrderId = payload.hostOrderId,
+                    hostName = payload.hostName,
+                    productLine = payload.productLine,
+                    orderType = payload.orderType,
+                    owner = payload.owner,
+                    receiver = payload.receiver,
+                    callbackUrl = payload.callbackUrl
```

Comment:
I think I mentioned how update order should have its own payload which only takes in fields that can be updated, it doesn't make sense to change hostOrderId, hostName, owner, or callbackUrl. The order update payload should only take in productline, orderType, and receiver, allowing users to only change these values.